### PR TITLE
Deployment Plans i3s API300 Support

### LINKED ISSUE #116: I3S - Deployment Plans API 300 Support
### Scenario/Intent
Use the Image Streamer - Deployment Plans resource for API 300 C7000/Synergy.

### Environment Details
 - **OneView SDK Version: Future
 - **OneView Appliance Version: 300
 - **OneView Client API Version: 300
 - **Ruby Version: 2.2.3+
 - **Platform: w/e

### Steps to Reproduce
Trying to run the resource of Image Streamer for API 300.

### Expected Result
Being able to call the resource methods and use it.

### Actual Result
It's currently not implemented.

### PR DESCRIPTION
**Pull request by aalexmonteiro**
### Description
Implementation of the deployment plans for i3s API300 support.

### Issues Resolved
Fixes #116 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/_client_i3s.rb.example [hunk 1]
```diff
@@ -15,4 +15,6 @@ require 'pry'
 puts "Connected to Image Streamer at #{@client.url}\n\n"
 
 # Instance variables for examples:
-@plan_script_name = 'Plan_Script_1'
```
Comment (tmiotto) [inline examples/_client_i3s.rb.example]: You renamed the `@plan_script_name` to `'Plan_Script_Name'` here, but `examples/image-streamer/api300/build_plan.rb` is still using it as if it was named `'Plan_Script_1'`

File: spec/integration/image-streamer/api300/deployment_plan/create_spec.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        oeBuildPlanURI: @build_plan['uri']
+      }
+
+      item = klass.new($client_i3s_300, options)
+      expect { item.create! }.not_to raise_error
+      item.retrieve!
+      expect(item['uri']).to be
+      expect(item['name']).to eq(options[:name])
+      expect(item['description']).to eq(options[:description])
+      expect(item['hpProvided']).to be options[:hpProvided]
```
Comment (tmiotto) [inline spec/integration/image-streamer/api300/deployment_plan/create_spec.rb]: I guess this was fixed in any of the already merged PR.

Please use `.to eq`

**Comment by aalexmonteiro** [comment]
@tmiotto , changes made.

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,8 +6,9 @@
 Added full support to Image Streamer Rest API version 300:
    - Build Plan
    - Deployment Plans
-   - Plan Scripts
+   - Golden Image
```
Comment (tmiotto) [inline CHANGELOG.md]: Please fix here.

All the entries should be in plural or all should be in singular.

**Comment by aalexmonteiro** [comment]
@tmiotto , done!

File: lib/oneview-sdk/image-streamer/resource/api300/deployment_plan.rb [hunk 4]
```diff
@@ -20,6 +20,16 @@ module API300
       #   bump, which may break your code.
```
Comment (marikrg) [inline lib/oneview-sdk/image-streamer/resource/api300/deployment_plan.rb]: I suggest revisit this note:

```ruby
       # @note This resource is unimplemented/unfinished at this point, so use at your own risk.
       #   This resource is subject to change drastically in the near future without a major version
        #   bump, which may break your code.
```

**Comment by aalexmonteiro** [comment]
@marikrg , done. I also removed the same note in other classes.

**Pull request merged**